test/integration/render_task_test.rb

```diff
@@ -0,0 +1,284 @@
+            args:
+            - echo
+            - log from pod1
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: config-for-pod1
+        data:
+          supports_partials: yep
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: independent-configmap
+        data:
+          value: renderer test
+        ---
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: pod2
+        spec:
+          restartPolicy: Never
+          activeDeadlineSeconds: 60
+          containers:
+          - name: pod2
+            image: busybox
+            args:
+            - echo
+            - log from pod2
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: config-for-pod2
+        data:
+          supports_partials: yep
+      RENDERED
+    end
+  end
+
+  def test_render_task_rendering_all_files
+    render = build_render_task(fixture_path('hello-cloud'))
+
+    assert_render_success render.run(mock_output_stream, [])
+    stdout_assertion do |output|
+      assert_match(/name: bare-replica-set/, output)
+      assert_match(/name: hello-cloud-configmap-data/, output)
+      assert_match(/name: ds-app/, output)
+      assert_match(/kind: PodDisruptionBudget/, output)
+      assert_match(/name: hello-job/, output)
+      assert_match(/name: redis/, output)
+      assert_match(/name: role-binding/, output)
+      assert_match(/name: resource-quotas/, output)
+      assert_match(/name: build-robot/, output)
+      assert_match(/name: stateful-busybox/, output)
+      assert_match(/name: hello-cloud-template-runner/, output)
+      assert_match(/name: unmanaged-pod-\w+/, output)
+      assert_match(/name: web/, output)
+    end
+  end
+
+  def test_render_task_multiple_templates_with_middle_failure
+    render = build_render_task(fixture_path('some-invalid'))
+    assert_render_failure render.run(mock_output_stream, [
+      'configmap-data.yml',
+      'yaml-error.yml',
+      'stateful_set.yml',
+    ])
+
+    stdout_assertion do |output|
+      assert_match(/name: hello-cloud-configmap-data/, output)
+      assert_match(/name: stateful-busybox/, output)
+    end
+
+    logging_assertion do |logs|
+      assert_match(/Invalid template: yaml-error.yml/, logs)
+    end
+  end
+
+  def test_render_invalid_binding
+    render = build_render_task(fixture_path('test-partials'), 'a': 'binding-a', 'b': 'binding-b')
+    fixture = 'deployment.yaml.erb'
+
+    assert_render_failure render.run(mock_output_stream, [fixture])
+    assert_logs_match_all([
+      /Invalid template: .*deployment.yaml.erb/,
+      /undefined local variable or method `supports_partials'/
+    ])
+  end
+
+  def test_render_invalid_arguments
+    render = build_render_task(fixture_path('test-partials'), 'a': 'binding-a')
+
+    assert_render_failure render.run(mock_output_stream, ["../"])
+    assert_logs_match_all([
```

Comment:
There's also `assert_logs_match` you can use for cases like this where you're only looking for one thing. That said, I often like to use `_match_all` instead and include surrounding headings in the assertion, to confirm it's printed _when_ I expected (right away vs in the summary, for example).